### PR TITLE
Changes to parser mediator to support EventDataStream #2956

### PR DESCRIPTION
**Pull request by joachimmetz**
Changes to parser mediator to support EventDataStream #2956

### CODE REVIEW

File: plaso/engine/worker.py [hunk 1]
```diff
@@ -218,22 +222,21 @@ def _AnalyzeFileObject(self, mediator, file_object):
 
       data = file_object.read(maximum_read_size)
 
-    display_name = mediator.GetDisplayName()
+    analyzer_attributes = {}
```
Comment (Onager) [inline plaso/engine/worker.py:225]: Remove this and directly set attributes on event_data_stream instead.

File: plaso/engine/worker.py [hunk 2]
```diff
@@ -122,25 +123,27 @@ def __init__(self, parser_filter_expression=None):
...
+    attributes that are added to produced event objects. Note that some
+    file systems allow directories to have data streams, such as NTFS.
 
     Args:
-      mediator (ParserMediator): mediates the interactions between
-          parsers and other components, such as storage and abort signals.
       file_entry (dfvfs.FileEntry): file entry whose data stream is to be
           analyzed.
       data_stream_name (str): name of the data stream.
+      display_name (str): human readable of the file entry currently being
```
[the remark this answers is not in the document]
Comment (joachimmetz) [reply plaso/engine/worker.py:139]: done

File: plaso/engine/worker.py (hunk 1 above)
Comment (joachimmetz) [reply plaso/engine/worker.py:233]: I have this as an after care point of https://github.com/log2timeline/plaso/issues/2956. I opt to do this in a different PR, to not change everything in a single PR.

**Comment by joachimmetz** [comment]
@Onager PR has been rebased and responded/addressed comments, PTAL

File: plaso/engine/worker.py (hunk 1 above)
Comment (Onager) [reply plaso/engine/worker.py:233]: I think your aftercare point is a bit different, I meant changing what's currently on like 241 to something like `setattr(event_data_stream, result.attribute_name, result.attribute_value)`. The aftercare point that I could identify from #2956 that I think you're talking about is `To do - determine if analyzers should set EventDataStream attributes directly?` which is a more involved change.
I'm just asking to remove the extra loop in this method.

Comment (joachimmetz) [reply plaso/engine/worker.py:233]: ah, yeah that is different than what I interpreted, I'll change it

Comment (joachimmetz) [reply plaso/engine/worker.py]: done

**Comment by joachimmetz** [comment]
@Onager PTAL

**Pull request merged**